kratos/includes/parallel_environment.h

```diff
@@ -105,6 +112,17 @@ class KRATOS_API(KRATOS_CORE) ParallelEnvironment
      */
     static FillCommunicator::Pointer CreateFillCommunicator(ModelPart& rModelPart);
 
+    /**
+     * @brief Create a Communicator object
+     * This method uses the previously registered communicator factory for the creation of a new communicator pointer
+     * @param rModelPart Model part required to retrieve the variables list from it
+     * @param rDataCommunicatorName Name of the data communicator to be retrieved for the communicator construction
+     * @return Communicator::UniquePointer Unique pointer to the new communicator
+     */
+    static Communicator::UniquePointer CreateCommunicator(
```

Comment:
I think the name of this fct should reflect that the deciding factor what it returns is if Kratos in general runs in MPI or not
(Same for the FillCommFactory btw)

Reply:
Note that the usage would be `ParallelEnvironment::CreateCommunicator`. We can rename it to something like `ParallelismAwareCreateCommunicator` but it might be a bit verbose and redundant.

Reply:
I would have suggested a name if I had a good suggestion :D

Reply:
I thought abt this some more, how about `CreateCommunicatorBasedOnGlobalParallelism`?
I really want to emphasize the `global`